crates/ruff_python_ast/src/python_version.rs

```diff
@@ -123,6 +123,15 @@ impl fmt::Display for PythonVersion {
     }
 }
 
+impl FromStr for PythonVersion {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let (major, minor) = s.split_once('.').ok_or(())?;
+        Self::try_from((major, minor)).map_err(|_| ())
```

Comment:
Nit: Not sure if it's worth it because it's so simple but this is somewhat duplicated now with the serde deserialization logic below.